labelmerge/workflow/rules/label_harmonization.smk

```diff
@@ -1,53 +1,81 @@
+
+        for candidate in dseg_files:
+            possible = True
+            candidate_entities = parse_file_entities(candidate)
+            del candidate_entities["extension"]
+            for key, val in candidate_entities.items():
+                if (key not in atlas_entities) or (atlas_entities[key] != val):
+                    possible = False
+            if possible:
+                if tsv_file is None:
+                    tsv_file = candidate
+                else:
+                    raise ValueError(
+                        "Multiple applicable metadata files applicable at this level "
+                        "of the directory hierarchy"
                     )
-                )
-            )
-        ),
-    # TODO: Update container that has appropriate dependencies
-    # container:
-    #     "docker://khanlab/neuroglia-core"
-    script:
-        "../scripts/label_split.py"
-
-
-def aggregate_input(wildcards):
-    checkpoint_output = checkpoints.split_labels.get(**wildcards).output.binary_dir
-    label_fname = Path(
-        bids(label="{label_idx}", suffix="mask.nii.gz", **wildcards)
-    ).name
-    label_mask_path = str(Path(checkpoint_output) / label_fname)
-    return expand(
-        label_mask_path,
-        label_idx=glob_wildcards(label_mask_path).label_idx,
+        # Move up a directory
+        cur_dir = cur_dir.parent
+
+    # If still no file found
+    if not tsv_file:
+        raise FileNotFoundError("No associated tsv file found")
+
+    return tsv_file
+
+
+def build_metadata_path(wildcards):
+    base_metadata = load_metadata(
+        Path(expand(base_inputs["labelmap"].input_path, **wildcards)[0]),
+        config["bids_dir"],
     )
+    overlay_metadata = load_metadata(
+        Path(expand(overlay_inputs["labelmap"].input_path, **wildcards)[0]),
+        config["overlay_bids_dir"],
+    )
+    return {"base_metadata": base_metadata, "overlay_metadata": overlay_metadata}
 
 
-rule aggregate:
+rule merge_labels:
     input:
-        aggregate_input,
+        unpack(build_metadata_path),
+        base_map=base_inputs["labelmap"].input_path,
+        overlay_map=overlay_inputs["labelmap"].input_path,
     output:
-        touch(
-            bids(
-                root=str(Path(config["output_dir"]) / "aggregated"),
-                suffix="aggregated",
-                **inputs["labelmap"].input_wildcards
-            )
+        merged_map=bids(
+            root=str(Path(config["output_dir"]) / "combined"),
+            suffix="dseg.nii.gz",
+            desc="combined",
+            **base_inputs["labelmap"].input_wildcards,
+        ),
+        merged_metadata=bids(
+            root=str(Path(config["output_dir"]) / "combined"),
+            suffix="dseg.tsv",
+            desc="combined",
+            **base_inputs["labelmap"].input_wildcards,
         ),
+    resources:
+        script=str(Path(workflow.basedir) / "scripts" / "labelmerge.py"),
     shell:
```

Comment:
Just a question, is the reason to pass this through as a script and run in shell so that `labelmerge.py` can also be called as a standalone if needed?

Reply:
Yep, more or less. I originally implemented it in its own repo for testing, but we could think about exposing it on its own in addition to the BIDS App CLI.